edge-hub/core/src/Microsoft.Azure.Devices.Edge.Hub.Core/config/EdgeHubConfigParser.cs

```diff
@@ -52,14 +62,63 @@ public static EdgeHubConfig GetEdgeHubConfig(EdgeHubDesiredProperties desiredPro
             return new ReadOnlyDictionary<string, RouteConfig>(routes);
         }
 
-        static Option<BrokerConfig> ParseBrokerConfig(BrokerProperties configuration)
+        Option<BrokerConfig> ParseBrokerConfig(BrokerProperties properties)
         {
-            if (configuration != null)
+            if (properties != null)
             {
-                throw new NotImplementedException();
+                return Option.Some(
+                    new BrokerConfig(
+                        this.ParseBridgeConfig(properties),
+                        this.ParseAuthorizationConfig(properties)));
             }
 
             return Option.None<BrokerConfig>();
         }
+
+        /// <summary>
+        /// EH Twin and policy definition in the Broker have different json schemas.
+        /// This method converts twin schema (BrokerProperties) into broker policy schema (AuthorizationConfig),
+        /// and validates it.
+        /// </summary>
+        Option<AuthorizationConfig> ParseAuthorizationConfig(BrokerProperties properties)
+        {
+            IList<string> errors = this.validator.ValidateAuthorizationConfig(properties.Authorizations);
+            if (errors.Count > 0)
+            {
+                string message = string.Join("; ", errors);
+                throw new InvalidOperationException($"Error validating authorization policy: {message}");
+            }
+
+            var result = new List<Statement>(properties.Authorizations?.Count ?? 0);
+            foreach (var statement in properties.Authorizations)
+            {
+                // parse deny rules first, since we agreed that they take precedence
+                // in case of conflicting rules.
+                foreach (var rule in statement.Deny)
+                {
+                    result.Add(new Statement(
+                        Effect.Deny,
+                        statement.Identities,
+                        rule.Operations,
+                        rule.Resources));
+                }
+
+                foreach (var rule in statement.Allow)
+                {
+                    result.Add(new Statement(
+                        Effect.Allow,
+                        statement.Identities,
+                        rule.Operations,
+                        rule.Resources));
+                }
+            }
+
+            return Option.Some(new AuthorizationConfig(result));
+        }
+
+        Option<BridgeConfig> ParseBridgeConfig(BrokerProperties properties)
+        {
+            return Option.None<BridgeConfig>();
```

Comment:
is this todo or it's supposed to always return none?

Reply:
This is todo. That's what I'll be doing next.

Reply:
Can you put TODO comment just so it doesn't get lost to the void (unless by doing next you mean in this PR of course)

Reply:
This is literally the next item on my list. I've already started, so it won't be lost :)